ts/components/messages/MessageCTABar.tsx

```diff
@@ -154,14 +174,27 @@ class MessageCTABar extends React.PureComponent<Props> {
 
   public render() {
     const { message, service, containerStyle, paymentByRptId } = this.props;
+    const { isSelectCalendarModalOpen } = this.state;
 
     const { due_date, payment_data } = message.content;
 
     if (due_date !== undefined || payment_data !== undefined) {
       return (
         <View style={[styles.mainContainer, containerStyle]}>
-          {due_date !== undefined &&
-            this.renderReminderCTA(due_date, message.content.subject)}
+          {due_date !== undefined && (
+            <React.Fragment>
+              {isSelectCalendarModalOpen && (
+                <SelectCalendarModal
+                  onCancel={this.onSelectCalendarCancel}
+                  onCalendarSelected={(calendar: Calendar) =>
+                    this.addReminderToCalendar(message, calendar, due_date)
```

Comment:
if the signature of `this.addReminderToCalendar` was:

```ts
addReminderToCalendar = (message, due_date) => (calendar) => void;
```

you would be able to simplify to:

```ts
onCalendarSelected={this.addReminderToCalendar(message, due_date)}
```

Reply:
Done